tst.c

```diff
@@ -81,9 +81,11 @@ static void *tst_del_word(tst_node **root,
     tst_node *victim = node;               /* begin deletion w/victim */
     tst_node *parent = tst_stack_pop(stk); /* parent to victim */
 
-    if (!victim->refcnt) {            /* if last occurrence */
-        if (!victim->key && freeword) /* check key is nul   */
-            free(victim->eqkid);      /* free string (data) */
+    if (!victim->refcnt) {              /* if last occurrence */
+        if (!victim->key && freeword) { /* check key is nul   */
+            free(victim->eqkid);        /* free string (data) */
+            victim->eqkid = NULL;
```

Comment:
It would make sense if `victim->eqkid` can be reclaimed. I would hesitate to apply the change. Any hints for violations?

Reply:
I don't get the point on what situation can we reclaimed the free pointer? Could you give me some brief explain?

Reply:
One of the motivation of `refcnt` ([reference counting](https://en.wikipedia.org/wiki/Reference_counting)) used in TST is to reclaim cycles. Thus, we don't have to rely on the value of `victim->eqkid` to decide its usage.